server/src/main.rs

```diff
@@ -1,6 +1,195 @@
 mod db;
+mod structs;
+
+#[macro_use]
+extern crate lazy_static;
+
+use data::{AuthorRole, AuthorStatus};
+use diesel::prelude::*;
+use diesel::r2d2::ConnectionManager;
+use std::convert::Infallible;
+use std::env::var;
+use structs::{CallbackQuery, TokenQuery, TokenResponse};
+
+use std::str::FromStr;
+use warp::{http::Uri, hyper::Response, Filter};
+
+lazy_static! {
+    static ref CLIENT_ID: String = var("GITHUB_CLIENT_ID").unwrap();
+    static ref CLIENT_SECRET: String = var("GITHUB_CLIENT_SECRET").unwrap();
+}
 
 #[tokio::main]
 async fn main() {
-    println!("Hello, world!");
+    let public = warp::get().and(warp::fs::dir("./server/public"));
+
+    let redirect_url = Uri::from_str(&format!(
+        "https://github.com/login/oauth/authorize?client_id={}&scope=read:org,read:user&state=random",
+        CLIENT_ID.to_string()
+    ))
+    .unwrap();
+
+    let database = crate::db::Pool::new(ConnectionManager::new(
+        "postgres://postgres@localhost/halfway-hackathon",
```

Comment:
```suggestion
        var("DATABASE_URL").expect("The `DATABASE_URL` environment variable is not set")
```
I think we should ideally initialize this from an environment variable for configurability, after which this should be good to merge.

Reply:
You're right, I forgot to remove that 🤦‍♂️